src/fosslight_util/download.py

```
@@ -612,10 +611,21 @@ def extract_compressed_file(fname, extract_path, remove_after_extract=True, comp
                 with contextlib.closing(tarfile.open(fname, "r:gz")) as t:
                     t.extractall(path=extract_path)
             else:
-                is_compressed_file = False
-                if compressed_only:
+                try:
+                    if zipfile.is_zipfile(fname):
+                        unzip(fname, extract_path)
+                    elif tarfile.is_tarfile(fname):
+                        with tarfile.open(fname) as tar:
+                            tar.extractall(extract_path)
```

Ruff (0.15.6)

[error] Uses of `tarfile.extractall()`

(S202)